config.go

```diff
@@ -551,31 +551,61 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
-func (s *Settings) GetGithubOauthString() (string, error) {
+func (s *Settings) GetGithubOauthStrings() ([]string, error) {
+	var tokens []string
+	var token_name string
+
 	token, ok := s.Credentials["github"]
 	if ok && token != "" {
-		return token, nil
+		// we want to make sure tokens[0] is always the default token
+		tokens = append(tokens, token)
+	} else {
+		return nil, errors.New("no 'github' token in settings")
 	}
 
-	return "", errors.New("no github token in settings")
+	for i := 1; i < 10; i++ {
+		token_name = fmt.Sprintf("github_alt%d", i)
+		token, ok := s.Credentials[token_name]
+		if ok && token != "" {
+			tokens = append(tokens, token)
+		} else {
+			break
+		}
+	}
+	return tokens, nil
 }
 
 func (s *Settings) GetGithubOauthToken() (string, error) {
 	if s == nil {
 		return "", errors.New("not defined")
 	}
 
-	oauthString, err := s.GetGithubOauthString()
+	oauthStrings, err := s.GetGithubOauthStrings()
 	if err != nil {
 		return "", err
 	}
+	timeSeed := time.Now().Nanosecond()
+	randomStartIdx := timeSeed % len(oauthStrings)
+	var oauthString string
+	for i := range oauthStrings {
+		oauthString = oauthStrings[randomStartIdx+i]
+		splitToken, err := splitToken(oauthString)
+		if err != nil {
+			grip.Error(errors.Wrapf(err, "problem with github_alt%d", i))
```

Comment:
In practice, logging errors to grip is unwieldy / unnecessary if we don't include some message fields to keep it organized.